week4/exercise.hs

```diff
@@ -0,0 +1,55 @@
+{-# OPTIONS_GHC -Wall #-}
+
+module Week4 where
+
+-- Exercise 1 : Wholemeal programming
+
+fun1' :: [Integer] -> Integer
+fun1' =  product . map (subtract 2) . filter even
+
+fun2' :: Integer -> Integer
+fun2' = sum . filter (even) . takeWhile (/=1) . iterate next
+  where
+    next :: Integer -> Integer
+    next x = if even x then div x 2 else 3 * x + 1
```

Comment:
Some people use infix version of the `div` function, like this:

```haskell
x `div` 2
```

But that depends on personal taste 🙂